lockup-contract/src/getters.rs

```diff
@@ -28,42 +27,40 @@ impl LockupContract {
-        if let TransfersInformation::TransfersEnabled {
-            transfers_timestamp,
-        } = &self.lockup_information.transfers_information
-        {
-            let lockup_timestamp = std::cmp::max(
-                transfers_timestamp
-                    .0
-                    .saturating_add(self.lockup_information.lockup_duration),
-                self.lockup_information.lockup_timestamp.unwrap_or(0),
-            );
-            let block_timestamp = env::block_timestamp();
-            if lockup_timestamp <= block_timestamp {
-                let unreleased_amount =
-                    if let &Some(release_duration) = &self.lockup_information.release_duration {
-                        let end_timestamp = lockup_timestamp.saturating_add(release_duration);
-                        if block_timestamp >= end_timestamp {
-                            // Everything is released
-                            0
-                        } else {
-                            let time_left = U256::from(end_timestamp - block_timestamp);
-                            let unreleased_amount = U256::from(lockup_amount) * time_left
-                                / U256::from(release_duration);
-                            // The unreleased amount can't be larger than lockup_amount because the
-                            // time_left is smaller than total_time.
-                            unreleased_amount.as_u128()
-                        }
-                    } else {
+
+        let lockup_timestamp = std::cmp::max(
+            self.lockup_information
+                .lockup_timestamp
+                .unwrap_or(0)
+                .saturating_add(self.lockup_information.lockup_duration),
+            self.lockup_information.lockup_timestamp.unwrap_or(0),
+        );
+
+        let block_timestamp = env::block_timestamp();
+
+        if lockup_timestamp <= block_timestamp {
+            let unreleased_amount =
+                if let &Some(release_duration) = &self.lockup_information.release_duration {
+                    let end_timestamp = lockup_timestamp.saturating_add(release_duration);
+                    if block_timestamp >= end_timestamp {
+                        // Everything is released
                         0
-                    };
+                    } else {
+                        let time_left = U256::from(end_timestamp - block_timestamp);
+                        let unreleased_amount =
+                            U256::from(lockup_amount) * time_left / U256::from(release_duration);
+                        // The unreleased amount can't be larger than lockup_amount because the
+                        // time_left is smaller than total_time.
+                        unreleased_amount.as_u128()
+                    }
+                } else {
+                    0
+                };
 
-                return unreleased_amount.into();
-            }
+            return unreleased_amount.into();
```

Comment:
We should actually clarify with HoS how the unlock works. Whether it's linear for `lockup_duration` when the user requested, or it's `lockup_duration` cliff with 100% unlock after.

Reply:
Current codebase has same principals, but different namings:
lock_total  = lock_near
lock_total for pending lock = end_unlock_near
lock_extra = lock_pending_near